### PR TITLE
Test discovery optimization.

### PR DESCRIPTION
**Pull request by pq**
* test call check short-circuiting to avoid needless Psi Element traversals
* minor cleanup

@stevemessick @jwren

### CODE REVIEW

File: src/io/flutter/dart/DartSyntax.java [hunk 1]
```diff
@@ -70,21 +70,20 @@ public static DartCallExpression findEnclosingFunctionCall(@NotNull PsiElement e
    *
    * @return true if the given element is a call to function, false otherwise
    */
-  public static boolean isCallToFunctionNamed(@Nullable PsiElement element, @NotNull String function) {
-    if (!(element instanceof DartCallExpression)) return false;
-    final String name = getCalledFunctionName((DartCallExpression)element);
+  public static boolean isCallToFunctionNamed(@NotNull DartCallExpression element, @NotNull String function) {
```
Comment (stevemessick) [inline src/io/flutter/dart/DartSyntax.java]: Doesn't this require a change to DartSuyntaxTest?

File: src/io/flutter/dart/DartSyntax.java [hunk 2]
```diff
@@ -70,21 +70,20 @@ public static DartCallExpression findEnclosingFunctionCall(@NotNull PsiElement e
...
    *
-   * @return true if the given element is a declaration, false otherwise
+   * @return true if the given element is a main declaration, false otherwise
    */
-  public static boolean isFunctionDeclarationNamed(@Nullable PsiElement element, @NotNull String name) {
+  public static boolean isMainFunctionDeclaration(@Nullable PsiElement element) {
     if (!(element instanceof DartFunctionDeclarationWithBodyOrNative)) return false;
     final String functionName = ((DartFunctionDeclarationWithBodyOrNative)element).getComponentName().getId().getText();
-    return Objects.equals(functionName, name);
+    return Objects.equals(functionName, "main");
```
Comment (jwren) [inline src/io/flutter/dart/DartSyntax.java]: nit: "main" could be a public static field (referenced anywhere else?)

File: src/io/flutter/FlutterUtils.java [hunk 3]
```diff
@@ -75,6 +76,19 @@ public static VirtualFile getRealVirtualFile(@Nullable PsiFile psiFile) {
     return psiFile != null ? psiFile.getOriginalFile().getVirtualFile() : null;
   }
 
+
+  /**
+   * Returns the Dart file for the given PsiElement, or null if not a match.
+   */
+  public static @Nullable DartFile getDartFile(final @Nullable PsiElement elt) {
```
Comment (jwren) [inline src/io/flutter/FlutterUtils.java]: Nit: Nullable annotation should be first token in signature, and parameter statement, like lines 57 and 73 above

File: src/io/flutter/dart/DartSyntax.java (hunk 1 above)
Comment (pq) [inline src/io/flutter/dart/DartSyntax.java]: No, but there should be more tests for sure.

Also, interestingly, I've got classpath issues and can't run these tests in EAP; will look into fixing that.

File: src/io/flutter/FlutterUtils.java (hunk 3 above)
Comment (pq) [inline src/io/flutter/FlutterUtils.java]: Updated.

File: src/io/flutter/dart/DartSyntax.java (hunk 2 above)
Comment (pq) [inline src/io/flutter/dart/DartSyntax.java]: In general we haven't been creating static fields for Strings that will just get interned (which is just to say there shouldn't be any significant performance difference).  As to whether it'd be clearer, I used to agree but now tend to think 

    Objects.equals(functionName, "main")

is clearer than

    Objects.equals(functionName, MAIN_FUNC_NAME)

where we have to define

    static final String MAIN_FUNC_NAME = "main";

elsewhere.

File: src/io/flutter/dart/DartSyntax.java (hunk 1 above)
Comment (pq) [inline src/io/flutter/dart/DartSyntax.java]: Test added.  Thanks! 👍

**Pull request merged**